src/mlpack/tests/split_data_test.cpp

```diff
@@ -123,6 +123,29 @@ BOOST_AUTO_TEST_CASE(SplitDataResultMat)
   CheckMatrices(input, concat);
 }
 
+BOOST_AUTO_TEST_CASE(SplitDataResultField)
+{
+  field<mat> input(1, 2);
+
+  mat matA(2, 10);
+  mat matB(2, 10);
+
+  size_t count = 0; // Counter for unique sequential values.
+  matA.imbue([&count]() { return ++count; });
+  matB.imbue([&count]() { return ++count; });
+
+  input(0, 0) = matA;
+  input(0, 1) = matB;
+
+  const auto value = Split(input, 0.5, false);
+  BOOST_REQUIRE_EQUAL(std::get<0>(value).n_cols, 1); // Train data.
+  BOOST_REQUIRE_EQUAL(std::get<1>(value).n_cols, 1); // Test data.
+
+  field<mat> concat = arma::join_rows(std::get<0>(value)(0), std::get<1>(value)(0));
+  // Order matters here.
+  CheckFields(input, concat);
+}
+
```

Comment:
This is probably a very bad way of writing test, could you lend a hand @rcurtin ?. For starters join_rows can't yield field type, how do i conteract that?

Reply:
I think the test strategy is ok; the only thing to work around would be the `join_rows` call.  You might consider creating `concat` as a `field` with size equal to the sum of sizes of `std::get<0>(value)` and `std::get<1>(value)`, then fill each element by iterating over everything in `std::get<0>(value)`, then iterating over everything in `std::get<1>(value)`.  It's a bit tedious but it should work. :+1:

Reply:
That's a good point, what if we use an initializer list instead? ```field<mat> concat = {std::get<0>(value)(0), std::get<1>(value)(0)};``` this works similar to ```arma::join_rows```.